### PR TITLE
Add sequence generators

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

**Comment by dcrissman** [comment]
How does this plug into the larger framework?

**Comment by bserdar** [comment]
Few more things are necessary:
 - Add a sequence generator extension to core
 - Add configuration support for sequence generators
 - Implement the sequence generator extension in MongoCRUDController, and
call the MongoSequenceGenerator
 - Add an interceptor, generate sequences there

**Comment by dcrissman** [comment]
So do you want this merged at this point?

**Comment by bserdar** [comment]
Yes, if it looks ok, pls merge.

File: crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoSequenceGenerator.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    private void initIndex() {
+        // Make sure we have a unique index on name
+        BasicDBObject keys=new BasicDBObject(NAME,1);
+        BasicDBObject options=new BasicDBObject("unique",1);
+        coll.ensureIndex(keys,options);
+    }
+   
+    public long getNextSequenceValue(String name,long init,long inc) {
```
Comment (dcrissman) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoSequenceGenerator.java]: I think this method could use some javadoc.

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoSequenceGenerator.java]: I'll add.

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoSequenceGenerator.java]: There are javadocs now.

**Pull request merged**